torch/nn/parallel/parallel_apply.py

```diff
@@ -8,31 +8,44 @@
     import Queue as queue
 
 
-def parallel_apply(modules, inputs):
+def parallel_apply(modules, inputs, kwargs):
     assert len(modules) == len(inputs)
+    if kwargs:
+        assert len(modules) == len(kwargs)
+
     # Fast track
     if len(modules) == 1:
-        return (modules[0](*inputs[0]),)
+        if kwargs is None:
+            return (modules[0](*inputs[0]),)
+        else:
+            return (modules[0](*inputs[0], **kwargs[0]),)
 
     lock = threading.Lock()
     results = {}
 
-    def _worker(module, input, results, lock):
+    def _worker(module, input, kwargs, results, lock):
         var_input = input
         while not isinstance(var_input, Variable):
             var_input = var_input[0]
         try:
             with torch.cuda.device_of(var_input):
-                output = module(*input)
+                if kwargs is not None:
+                    output = module(*input, **kwargs)
+                else:
+                    output = module(*input)
             with lock:
                 results[input] = output
         except Exception as e:
             with lock:
                 results[input] = e
-
-    threads = [threading.Thread(target=_worker,
-                                args=(module, input, results, lock))
-               for module, input in zip(modules, inputs)]
+    if kwargs is None:
+        threads = [threading.Thread(target=_worker,
+                                    args=(module, input, kwargs, results, lock))
+                   for module, input in zip(modules, inputs)]
+    else:
+        threads = [threading.Thread(target=_worker,
+                                    args=(module, input, kwargs, results, lock))
+                   for module, input, kwargs in zip(modules, inputs, kwargs)]
```

Comment:
it is possible to avoid this complication of kwargs by predefining `module` functions as `partial` functions, might be cleaner